wrapper/src/main/java/software/amazon/jdbc/hostlistprovider/monitoring/ClusterTopologyMonitorImpl.java

```diff
@@ -438,13 +441,58 @@ public void run() {
     }
   }
 
+  protected void shutdownNodeExecutorService() {
+    if (this.nodeExecutorService != null) {
+
+      this.nodeExecutorLock.lock();
+      try {
+
+        if (this.nodeExecutorService == null) {
+          return;
+        }
+
+        if (!this.nodeExecutorService.isShutdown()) {
+          this.nodeExecutorService.shutdown();
+        }
+
+        try {
+          if (!this.nodeExecutorService.awaitTermination(30, TimeUnit.SECONDS)) {
+            this.nodeExecutorService.shutdownNow();
+          }
+        } catch (InterruptedException e) {
+          // do nothing
```

Comment:
Would it make sense to call `this.nodeExecutorService.shutdownNow();` here if the awaitTemrmination request is interrupted

Reply:
There's another final block in run() method that can catch InterruptedException and handle it.